californium-core/src/main/java/org/eclipse/californium/core/network/UdpMatcher.java

```diff
@@ -478,14 +478,13 @@ private boolean isResponseRelatedToRequest(final Exchange exchange, final Correl
 			// no correlation information available for request, thus any
 			// additional correlation information available in the response is ignored
 			return true;
-		} else if (exchange.getCorrelationContext() instanceof DtlsCorrelationContext) {
+		} else if (exchange.getCorrelationContext().get(DtlsCorrelationContext.KEY_SESSION_ID) != null) {
```

Comment:
Is there a particular reason why you would like to get rid of the *instanceof* check? You are still depending on the `DtlsCorrelationContext` class. The only reason I can see why you would want to do this is because you expect to find a DTLS correlation context in the exchange that is not a `DtlsCorrelationContext`. However, since Californium uses Element-Connector and Scandium, I wonder where such a correlation context should come from...